### PR TITLE
Unit tests and clean up

### LINKED ISSUE #4: Add unit tests

### LINKED ISSUE #9: Properly decode longlongs
Currently, when reading integers, we assume the incoming payload will be encoded as a `long` or a  `ulong`. Technically, they can also be encoded as `longlong` or `ulonglong`, but we do not handle that case.

### PR DESCRIPTION
**Pull request by z-silver**
Adds some unit tests and cleans up a few things.

Closes #4. Closes #9.

### CODE REVIEW

File: examples/hello/hello.zig [hunk 1]
```diff
@@ -21,7 +21,7 @@ pub fn main() !void {
     try stderr.print("Connected.\n", .{});
 
     const self = try node.self();
-    const message = zerl.With_Pid([]const u8){ self.*, "Hello, World!" };
+    const message: zerl.With_Pid([]const u8) = .{ self.*, "Hello, World!" };
```
Comment (EduardoLR10) [inline examples/hello/hello.zig:24]: Nice!

File: src/Decoder.zig [hunk 2]
```diff
@@ -40,133 +39,245 @@ pub const Error = std.mem.Allocator.Error || error{
...
+    try erl.encoder.write_any(&buf, point);
+
+    const decoder: Decoder = .{
+        .buf = &buf,
+        .index = &index,
+        .allocator = testing.failing_allocator,
+    };
+
+    try testing.expectEqual(point, decoder.parse_tuple(Point));
+}
```
Comment (EduardoLR10) [inline src/Decoder.zig:126]: Is it common practice in Zig for tests to live in the same place as the implementation? This is the case in Rust, and I don't like it.

File: src/Decoder.zig [hunk 3]
```diff
@@ -40,133 +39,245 @@ pub const Error = std.mem.Allocator.Error || error{
...
+    var value: T = undefined;
     inline for (&value) |*elem| {
         elem.* = try self.parse(@TypeOf(elem.*));
     }
     return value;
 }
 
+test parse_tuple {
+    // We need the comptime field here in 0.13.0 because of a bug.
+    // Problem is fixed in zig master.
```
Comment (EduardoLR10) [inline src/Decoder.zig:108]: Nice catch!

File: src/Decoder.zig [hunk 4]
```diff
@@ -40,133 +39,245 @@ pub const Error = std.mem.Allocator.Error || error{
...
                 }
-                present_fields.set(idx);
-                counter += 1;
-            }
+                present_keys.insert(key);
+            },
         }
     }
-    if (size < counter) return error.too_few_map_entries;
+    assert(size == present_keys.count());
```
Comment (EduardoLR10) [inline src/Decoder.zig:162]: Nice! The loop was indeed avoidable!

File: src/Decoder.zig [hunk 5]
```diff
@@ -180,66 +291,153 @@ fn parse_float(self: Decoder, comptime T: type) Error!T {
...
+    if (max_name_length < atom_size) return error.could_not_decode_enum;
+
+    var atom_name: [max_name_length + 1]u8 = undefined;
+    try erl.validate(
+        error.decoding_atom,
+        ei.ei_decode_atom(self.buf.buff, self.index, &atom_name),
+    );
+    const name = atom_name[0..@as(c_uint, @intCast(atom_size))];
+    return tag_map.get(name) orelse error.could_not_decode_enum;
+}
```
Comment (EduardoLR10) [inline src/Decoder.zig:361]: Nice new checks! I really didn't know about the length limitation.

**Review by EduardoLR10 — APPROVED** [review]

File: src/Decoder.zig (hunk 2 above)
Comment (z-silver) [reply src/Decoder.zig:126]: It is, actually. We don't have to do it, necessarily, but it's the path of least resistance and so it's what most people do.

**Pull request merged**